kubectl-plugin/pkg/cmd/cluster/get.go

```diff
@@ -72,19 +76,12 @@ func (options *ClusterOptions) Validate() error {
 	if len(options.args) > 1 {
 		return fmt.Errorf("too many arguments, either one or no arguments are allowed")
 	}
-
 	return nil
 }
 
-func (options *ClusterOptions) Run(ctx context.Context) error {
-	// Retrieves the REST client based on .kubeconfig
-	restConfig, err := options.configFlags.ToRESTConfig()
-	if err != nil {
-		return err
-	}
-
-	// Retrieves the HTTPClient with the provided Rest client
-	dynamicClient, err := dynamic.NewForConfig(restConfig)
+func (options *ClusterOptions) Run(ctx context.Context, factory cmdutil.Factory) error {
+	// Retrieves the dynamic client with factory.
+	dynamicClient, err := factory.DynamicClient()
```

Comment:
Out of curiosity, what's a "dynamic client"?

Reply:
It provides interaction with non-standard Kubernetes resources so like CRDs and potentially unknown resources.

Reply:
Thank you for the explanation!